### ISSUE TITLE
The react component should optionally pass in the container size

### ISSUE THREAD

**Issue by ZeeCoder** [ZeeCoder/container-query#82]
Since the react component observes size changes on the root element already,
it might as well inject the width / height as props.

Probably passing in a `size` prop optionally is most straight-forward.
(The way [react-sizeme does it](https://github.com/ctrlplusb/react-sizeme) does it.)

Although, making the package use the render-prop pattern is tempting too 🤔.
```jsx
render() {
  <ContainerQuery stats={stats} render={size => ... } />
}
```
☝ This would allow the user to alias the prop however they like, and makes it crystal clear what's happening.

**Comment by ZeeCoder** [comment]
I think I'll keep both, so then it fits in the next minor release:
```js
import withContainerQuery, { ContainerQuery } from '@zeecoder/react-container-query';
```
then:
- use `withContainerQuery()` for simple use-cases where size doesn't matter in JS-land,
- use `<ContainerQuery>` if size is needed to decide what to render.

**Comment by ZeeCoder** [comment]
Might as well expose a <ResizeObserver> too I guess:
```jsx
<ResizeObserver render={size => ... }>
```

**Comment by ZeeCoder** [comment]
Fixed by 2.4.0
